index.js

```diff
@@ -338,14 +338,13 @@ DailyRotateFile.prototype.query = function (options, callback) {
         results = results.reverse();
       }
 
-      if(createdFiles.length)
+      if (createdFiles.length) {
         readNextFile(createdFiles.shift());
-      if (callback) {
+      } else if (callback) {
```

Comment:
Should this actually be an `else if`? The original patch did not appear to follow that path.

Reply:
Sorry, yes it should be an else if.

Reply:
About the linting errors, i ran npm test as you suggested, i corrected all the errors i got. but there's a TODO comment at line 290:
  290:3  warning  Unexpected 'todo' comment  no-warning-comments
Since i didn't write that comment, i dont know if it's safe to edit/remove it.

Reply:
Seems like some of the travis failures may be due to the use of the let keyword (https://travis-ci.org/winstonjs/winston-daily-rotate-file/jobs/275100126). Can you update to remove those so that we can continue supporting the older nodejs versions?

Thanks!

Reply:
ok, i'll replace let with var